### PR TITLE
fix vulkan crash while runing in background

### PR DESCRIPTION
**Pull request by bofeng-song**

### CODE REVIEW

File: native/cocos/renderer/gfx-vulkan/VKSwapchain.cpp [hunk 1]
```diff
@@ -258,6 +258,9 @@ void CCVKSwapchain::doResize(uint32_t width, uint32_t height, SurfaceTransform /
 }
 
 bool CCVKSwapchain::checkSwapchainStatus(uint32_t width, uint32_t height) {
+    if (_gpuSwapchain->vkSurface == VK_NULL_HANDLE) { // vkSurface will set to VK_NULL_HANDLE after call doDestroySurface
```
Comment (minggo) [inline native/cocos/renderer/gfx-vulkan/VKSwapchain.cpp]: will set -> will be set.

Why this function will be invoked after surface is destroyed.

**Review by minggo — APPROVED** [review]

File: native/cocos/renderer/gfx-vulkan/VKSwapchain.cpp (hunk 1 above)
Comment (bofeng-song) [reply native/cocos/renderer/gfx-vulkan/VKSwapchain.cpp]: called from ts

**Review by minggo — APPROVED** [review]

**Comment by star-e** [comment]
Is this problem related to https://github.com/cocos/cocos-engine/pull/10776 ?

**Comment by bofeng-song** [comment]
> Is this problem related to #10776 ?

No, #10776 is trigger by restart Application, but this pr fix running at background.

**Comment by minggo** [comment]
Oh, sorry, it is my mistake. I thought @star-e agree with it.
@star-e if you do not agree with it, i can revert it.

**Comment by star-e** [comment]
I'm not very familiar with Android surface and swapchain. I think this fix is ok. @minggo

**Comment by minggo** [comment]
Ok, thanks.

**Pull request merged**